core/app/c/[communitySlug]/forms/[formSlug]/edit/actions.ts

```diff
@@ -0,0 +1,37 @@
+"use server";
+
+import { logger } from "logger";
+
+import { db } from "~/kysely/database";
+import { isUniqueConstraintError } from "~/kysely/errors";
+import { getLoginData } from "~/lib/auth/loginData";
+import { autoRevalidate } from "~/lib/server/cache/autoRevalidate";
+import { defineServerAction } from "~/lib/server/defineServerAction";
+
+export const updateForm = defineServerAction(async function updateForm({ communityId, name }) {
+	const user = await getLoginData();
+
+	if (!user) {
+		return {
+			error: "Not logged in",
+		};
+	}
+	try {
+		const formUpdateQuery = await autoRevalidate(
+			db
+				.updateTable("forms")
+				.set({ name: name })
+				.where("communityId", "=", communityId)
+				.returning("id")
+		);
+		//
+		await formUpdateQuery.execute();
+	} catch (error) {
+		if (isUniqueConstraintError(error)) {
+			const column = error.constraint === "forms_slug_key" ? "slug" : "name";
+			return { error: `A form with this ${column} already exists. Choose a new ${column}` };
+		}
+		logger.error({ msg: "error creating form", error });
```

Comment:
This log message isn't correct. I also think this is a situation where we should be explicitly reporting the error to sentry, since it means something unexpected happened. It's ok if you don't do this now, but it would be really nice if we had a utility function that both called logger.error and Sentry.captureException and then used it here (and in many other places!)

Reply:
```typescript
export function logError(message: string, error: Error) {
	logger.error({ message, error });
	Sentry.captureException(error);
}

``` 

i have added this. I dont really know how fancy it should be but it does the two things together now.